src/NHibernate/Driver/IDriver.cs

```diff
@@ -158,5 +158,10 @@ public interface IDriver
 		/// crash due to this, because they re-use the connection in the second phase.
 		/// </remarks>
 		bool HasDelayedDistributedTransactionCompletion { get; }
+
+		/// <summary>
+		/// The minimal date supported by this driver.
+		/// </summary>
+		DateTime MinDate { get; }
```

Comment:
What about other drivers?

The datetime range for [MySQL](https://dev.mysql.com/doc/refman/5.7/en/datetime.html), [Oracle](https://docs.oracle.com/cd/B19306_01/olap.102/b14346/dml_datatypes005.htm)  is 1000-01-01 to 9999-12-31

For [Postgres](https://www.postgresql.org/docs/current/static/datatype-datetime.html) is 4713 BC to 294276 AD

Reply:
I have just put .Net `MinValue` and let the test tell me if it was ok. So that is strange for MySql, and for Postgres, as .Net native type does not allow BC dates, it is limited by .Net rather than by the DB, but that still mean a limit at `MinValue`.

Reply:
According to this [other doc](https://docs.oracle.com/database/122/SQLRF/Data-Types.htm#GUID-5405B652-C30E-4F4F-9D33-9A4CB2110F1B), Oracle `DATE` support date down to 4712 BC. So I think we can keep `MinDate` for Oracle.

About MySql, they also write:
> For the DATE and DATETIME range descriptions, “supported” means that although earlier values might work, there is no guarantee.

So it just happens to work in our case, but we should not complain if it ceases too. I will set its `MinDate` accordingly.